src/app/qgsmaptoolcircle2tangentspoint.h

```diff
@@ -0,0 +1,59 @@
+/***************************************************************************
+    qgsmaptoolcircle2tangentspoint.h  -  map tool for adding circle
+    from 2 tangents and a point
+    ---------------------
+    begin                : July 2017
+    copyright            : (C) 2017 by Loïc Bartoletti
+    email                : lbartoletti at tuxfamily dot org
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 3 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSMAPTOOLCIRCLE2TANGENTSPOINT_H
+#define QGSMAPTOOLCIRCLE2TANGENTSPOINT_H
+
+#include "qgspointlocator.h"
+#include "qgsmaptooladdcircle.h"
+#include "qspinbox.h"
+
+class QSpinBox;
+
+class QgsMapToolCircle2TangentsPoint: public QgsMapToolAddCircle
+{
+    Q_OBJECT
+
+  public:
+    QgsMapToolCircle2TangentsPoint( QgsMapToolCapture *parentTool, QgsMapCanvas *canvas, CaptureMode mode = CaptureLine );
+    ~QgsMapToolCircle2TangentsPoint();
+
+    void cadCanvasReleaseEvent( QgsMapMouseEvent *e ) override;
+    void cadCanvasMoveEvent( QgsMapMouseEvent *e );
+
+  public slots:
+    void radiusSpinBoxChanged( int radius );
+
+  private:
+    //! Return the point where segments are intersected. Method from QgsGeometryUtils doesn't work for special cases used by this tool.
```

Comment:
do you think the QgsGeometryUtils method should be fixed?

Reply:
Not at the moment. I have launched a QEP to improve editing tools and some of these will need this kind of method. The QgsGeometryUtils' method is used for the topology. I have found (at the moment) only one case where there is a problem (for this intersection): when lines are continuous.A rewrite can will be done by merging this needs and the other editing tools.